lib/reservation.rb

```diff
@@ -0,0 +1,113 @@
+require 'awesome_print'
+require 'date'
+require 'csv'
+require_relative 'room'
+
+module Hotel
+  class Reservation
+    attr_reader :reservation_id, :check_in, :check_out, :number_of_days
+    attr_accessor :status, :room_number, :room
+    VALID_STATUS = [:pending, :denied, :confirmed]
+
+    def initialize(reservation_id, check_in, number_of_days, status = :pending, room = nil, room_number = nil)
+      @reservation_id = reservation_id
+      
+      if check_in.class != Date
+        raise ArgumentError.new("Invaid, check-in date must be Date.")
+      end
+      @check_in = check_in
+
+
+      if number_of_days <= 0
+        raise ArgumentError.new("Invalid, number of days must be greater than 0")
+      end
+      @number_of_days = number_of_days
+      
+      @status = status
+      @check_out = check_in + number_of_days 
+
+      if room
+        @room = room
+        @room_number = room.room_number
+      else
+        @room = room
+        @room_number = room_number
+      end
+      
+    end #initialize
+    #****************************************************************
+
+    #Total cost for a given reservation
+    #returns total_cost (float)
+    def total_cost()
+      total_cost = nil
+      if @room != nil
+        total_cost = @room.price.to_f * @number_of_days
+      end
+      return total_cost
+    end  
+    #****************************************************************
+    
+    # checks if aDate is within the check_in and check_out
+    # aDate (Date)
+    # returns true or false 
+    def is_within_stay(aDate)
```

Comment:
Variables (even arguments) should be named using `snake_case`:

```suggestion
    def is_within_stay(a_date)
```